src/plugins/hooks.js

```diff
@@ -160,26 +165,35 @@ module.exports = function (Plugins) {
 		}
 		async.each(hookList, function (hookObj, next) {
 			if (typeof hookObj.method === 'function') {
-				var timedOut = false;
-
-				var timeoutId = setTimeout(function () {
+				let timedOut = false;
+				const timeoutId = setTimeout(function () {
 					winston.warn('[plugins] Callback timed out, hook \'' + hook + '\' in plugin \'' + hookObj.id + '\'');
 					timedOut = true;
 					next();
 				}, 5000);
 
-				try {
-					hookObj.method(params, function () {
-						clearTimeout(timeoutId);
-						if (!timedOut) {
-							next.apply(null, arguments);
-						}
-					});
-				} catch (err) {
+				const onError = function (err) {
 					winston.error('[plugins] Error executing \'' + hook + '\' in plugin \'' + hookObj.id + '\'');
 					winston.error(err);
 					clearTimeout(timeoutId);
 					next();
+				};
+				const callback = function () {
```

Comment:
Suggest using `...args` here as well.

Also, define these two functions normally

Reply:
Ok about `...args` but for the functions declaration, looks like there is a lint rule to move the functions up if I don't assign them to a `const`

```
error  Move function declaration to function body root  no-inner-declarations
```

 and my argument not to do that is to keep relevant code close and a meaningful order

Reply:
In that case I recommend using arrow functions instead.